google_reauth/challenges.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import base64
+import sys
+
+import pyu2f.convenience.authenticator
+import pyu2f.errors
+import pyu2f.model
+import six
+
+from google_reauth import _helpers
+
+
+REAUTH_ORIGIN = 'https://accounts.google.com'
+
+
+def build_challenges():
+    """Returns ."""
+    out = {}
+    for c in [SecurityKeyChallenge(),
+              PasswordChallenge()]:
+        if c.is_locally_eligible():
+            out[c.get_name()] = c
+    return out
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ReauthChallenge(object):
+    """Base class for reauth challenges."""
+
+    @abc.abstractmethod
+    def get_name(self):
+        """Returns the name of the challenge."""
+        pass
+
+    @abc.abstractmethod
+    def is_locally_eligible(self):
+        """Returns true if a challenge is supported locally on this machine."""
+        pass
+
+    @abc.abstractmethod
+    def obtain_credentials(self, metadata):
```

Comment:
`credentials` is an insanely overloaded term in auth, can you be specific here and use `obtain_challenge_credentials`?